src/services/decoratelayerloading.js

```diff
@@ -0,0 +1,79 @@
+goog.provide('ngeo.DecorateLayerLoading');
+
+goog.require('goog.asserts');
+goog.require('ngeo');
+
+
+/**
+ * Provides a function that adds a 'loading 'property (using
+ * `Object.defineProperty`) to the layer.
+ * This property is true when the layer is loading and false otherwise.
+ *
+ * Example:
+ *
+ *      <span ng-if="layer.loading">please wait</span>
+ *
+ * @typedef {function(ol.layer.Base, angular.Scope)}
+ * @ngdoc service
+ * @ngname ngeoDecorateLayerLoading
+ */
+ngeo.DecorateLayerLoading;
+
+
+/**
+ * @param {ol.layer.Base} layer Layer to decorate.
+ * @param {angular.Scope} $scope Scope.
+ */
+ngeo.decorateLayerLoading = function(layer, $scope) {
+  goog.asserts.assertInstanceof(layer, ol.layer.Base);
+
+  var sources = [];
+  if (layer instanceof ol.layer.Group) {
+    // layer group
+    sources = layer.getLayersArray().map(function(layer) {
+      goog.asserts.assert(layer instanceof ol.layer.Layer);
+      return layer.getSource();
+    });
+  } else {
+    goog.asserts.assert(layer instanceof ol.layer.Layer);
+    sources = [layer.getSource()];
+  }
+
+  layer.set('load_count', 0);
```

Comment:
Have you considered setting the property silently here and in the rest of the file?